Sample_Notebooks/RFEP_analysis.ipynb

```diff
@@ -0,0 +1,175 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load_ext autoreload\n",
+    "%autoreload 2\n",
+    "\n",
+    "import safep\n",
+    "from glob import glob\n",
+    "import re\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as plt\n",
+    "import os"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# This should be the only user parameter we need\n",
+    "logfile = 'RFEP_decouple.log'\n",
+    "\n",
+    "# Everything is automated and generic below this"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "global_conf, colvars, biases, TI_traj = safep.parse_Colvars_log(logfile)\n",
+    "\n",
+    "\n",
+    "# Need to pick the right restraint if there are several\n",
+    "# TODO look for harmonic wall with changing k\n",
+    "\n",
+    "for b in biases:\n",
+    "    if float(b['targetForceConstant']) >= 0 or b['decoupling'] in ['on', 'yes', 'true']:\n",
+    "        restraint_config = b\n",
+    "        break\n",
+    "\n",
+    "rest_name = restraint_config['name']\n",
+    "cvs = restraint_config['colvars']\n",
+    "\n",
+    "print(f'Processing TI data for restraint {rest_name} on CVs {cvs}')\n",
+    "\n",
+    "\n",
+    "path = os.path.dirname(logfile) # We assume the colvars traj and log are in the same directory\n",
+    "\n",
+    "if 'traj_file' in global_conf:\n",
+    "    colvarsTraj = path + global_conf['traj_file']\n",
+    "else:\n",
+    "    colvarsTraj = path + global_conf['output_prefix'] + '.colvars.traj'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Extract precomputed TI info from traj file\n",
+    "dAdL = TI_traj[rest_name]['dAdL']\n",
+    "lambdas =  TI_traj[rest_name]['L']\n",
+    "\n",
+    "# if lambdaExponent >=2, set a zero derivative for lambda=0 (might be NaN in the data)\n",
+    "if int(restraint_config['lambdaExponent']) >= 2 and np.isnan(dAdL[-1]):\n",
+    "    dAdL[-1] = 0.0\n",
+    "\n",
+    "if restraint_config['decoupling']: # lambdas have opposite meaning\n",
+    "    # Convert to coupling\n",
+    "    dAdL = - np.array(dAdL)\n",
+    "    lambdas = 1.0 - np.array(lambdas)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Setup and processing of colvars data\n",
```

Comment:
Is this boilerplate that should be wrapped up into a function? (a later PR is fine, just so I know it should be made an issue)

Reply:
Definitely! A lot of this code should become SAFEP functions, and possibly a single user-visible one, as suggested in the PR description.